### PR TITLE
feat: add Java API for CSS utility classes

### LINKED ISSUE #2410: Create Java API for Lumo Utility Classes
### Describe your motivation

Consider creating a Java API for Lumo Utility Classes. Currently, if one wants to make use of those classes in a Flow app, one would have to do something like
```
addClassNames("flex", "flex-col", "gap-l", "mb-s", "mt-0", "text-s");
```
But this is suboptimal because (1) users could make typos in the String literals, (2) users don't necessarily know all the available class names, and (3) users can unknowingly add classess that cancel other classes (like adding `border-0` followed by `border`). 

### Describe the solution you'd like

I suggest adding `addUtilityClasses(UtilityClass)` and `removeUtilityClasses(UtilityClass)` methods to `com.vaadin.flow.component.HasStyle` interface (where `UtilityClass` is an Enum).

### PR DESCRIPTION
**Pull request by anezthes**
Fixes #2410 

See https://github.com/vaadin/platform/issues/2784

### CODE REVIEW

**Comment by rolfsmeds** [comment]
As the main issue with using string literals in Java is discoverability and code legibility, I feel that we should strive to maximize those with this API, and I'm a bit worried about how these non-nested classes will affect that. Some examples how this might be an issue:

1. Not easy to get an autocomplete list of all the available constants, or top-level categories of constants (unless you type the package as prefix).

2. Not easy to identify and understand constants in code:
Let's say you see this in your codebase:
`addClassName(Display.HIDDEN);`
It's not very clear what that `Display` thing is.

3. Potential confusion with other classes:
Typing e.g. Grid gives you an autocomplete dropdown like this (example is IntelliJ but same applies to other IDEs) with utility-class-constants and other similarly named classes mixed (ignore the package name of the utility-class-classes, I just quickly pasted them to a project for testing):
image

All of these issue would be solved by nesting the utility class classes (ugh) inside the `Lumo` class, or some new class like `LumoUtility` or `LumoStyles`. That would get rid of the autocomplete issues since you'd type `LumoStyles.`. to get the list of constant categories, and `LumoStyles.Grid` to get the css-grid constants without Grid-the-component stuff,  and code where they're used would be easier to grok:

`addClassName(LumoStyles.Display.HIDDEN);`

**Comment by TatuLund** [comment]
> All of these issue would be solved by nesting the utility class classes (ugh) inside the Lumo class, or some new class like LumoUtility or LumoStyles. 

For reference this was also case in Vaadin 8, where there is ValoTheme class wrapping the constans in similar fashion.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 1]
```diff
@@ -0,0 +1,1426 @@
...
+        public static final String START = "items-start";
+        public static final String STRETCH = "items-stretch";
+
+        private AlignItems() {
+        }
+
+    }
+
+    /**
+     * Classes for overriding individual items align-item property. Applies to
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:79]:
```suggestion
     * Classes for overriding individual items' align-item property. Applies to
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 2]
```diff
@@ -0,0 +1,1426 @@
...
+        public static final String SUCCESS_50 = "bg-success-50";
+        public static final String SUCCESS_10 = "bg-success-10";
+
+        private Background() {
+        }
+
+    }
+
+    /**
+     * Border related classes.
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:134]:
```suggestion
     * Border-related classes.
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 3]
```diff
@@ -0,0 +1,1426 @@
...
+        public static final String LARGE = "h-l";
+        public static final String XLARGE = "h-xl";
+
+        private Height() {
+        }
+
+    }
+
+    /**
+     * Classes for defining the size of elements working as icons.
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:870]:
```suggestion
     * Classes for defining the size of elements used as icons.
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 4]
```diff
@@ -0,0 +1,1426 @@
...
+        public static final String FULL = "min-w-full";
+        public static final String NONE = "min-w-0";
+
+        private MinWidth() {
+        }
+
+    }
+
+    /**
+     * Classes for setting the overflow behaviour of an element.
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:1145]:
```suggestion
     * Classes for setting the overflow behavior of an element.
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 5]
```diff
@@ -0,0 +1,1426 @@
...
+            public static final String MEDIUM = "pe-m";
+            public static final String LARGE = "pe-l";
+            public static final String XLARGE = "pe-xl";
+
+            private End() {
+            }
+        }
+
+        /**
+         * Classes for setting both the right and the left paddings of an
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:1208]:
```suggestion
         * Classes for setting both the right and left paddings of an
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 6]
```diff
@@ -0,0 +1,1426 @@
...
+            public static final String MEDIUM = "pt-m";
+            public static final String LARGE = "pt-l";
+            public static final String XLARGE = "pt-xl";
+
+            private Top() {
+            }
+        }
+
+        /**
+         * Classes for defining both the vertical and the horizontal paddings of
```
Comment (dougjewsbury) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:1291]:
```suggestion
         * Classes for defining both the vertical and horizontal padding of
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 7]
```diff
@@ -0,0 +1,1426 @@
...
+        }
+
+    }
+
+    /**
+     * Classes for setting the text overflow.
+     */
+    public static final class TextOverflow {
+
+        public static final String CLIP = "overflow-cip";
```
Comment (web-padawan) [inline vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java:1370]: Typo: `cip` -> `clip`

```suggestion
        public static final String CLIP = "overflow-clip";
```

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 1 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java [hunk 8]
```diff
@@ -0,0 +1,1426 @@
...
+    private LumoUtility() {
+    }
+
+    /**
+     * Accessibility related classes.
+     */
+    public static final class Accessibility {
+
+        /**
+         * Hides content visually while remaining it available to screen
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 2 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 3 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 4 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 5 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 6 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

File: vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java (hunk 7 above)
Comment (DiegoCardoso) [reply vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/LumoUtility.java]: Done.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**